diff -up flightgear-2020.3.6/CMakeLists.txt.omv~ flightgear-2020.3.6/CMakeLists.txt
--- flightgear-2020.3.6/CMakeLists.txt.omv~	2021-01-25 03:26:45.595926609 +0000
+++ flightgear-2020.3.6/CMakeLists.txt	2021-01-25 03:27:39.208323010 +0000
@@ -512,7 +512,11 @@ if (CMAKE_CXX_COMPILER_ID STREQUAL "Clan
     set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -g -DNDEBUG")
     set(CMAKE_C_FLAGS_RELWITHDEBINFO  "-O3 -g -DNDEBUG")
 
-    set(SIMD_COMPILER_FLAGS "-msse2 -mfpmath=sse -ftree-vectorize -ftree-slp-vectorize")
+    if (X86 OR X86_64)
+      set(SIMD_COMPILER_FLAGS "-msse2 -mfpmath=sse -ftree-vectorize -ftree-slp-vectorize")
+    else()
+      set(SIMD_COMPILER_FLAGS "-ftree-vectorize -ftree-slp-vectorize")
+    endif()
 
   #set(CMAKE_XCODE_ATTRIBUTE_CLANG_CXX_LIBRARY "libc++")
    #set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -stdlib=libc++")
