diff -up flightgear-2020.3.18/CMakeLists.txt.omv~ flightgear-2020.3.18/CMakeLists.txt
--- flightgear-2020.3.18/CMakeLists.txt.omv~	2023-06-20 04:49:16.021173330 +0200
+++ flightgear-2020.3.18/CMakeLists.txt	2023-06-20 04:49:23.079229803 +0200
@@ -56,7 +56,7 @@ endif()
 
 # let's use & require C++11 - note these are only functional with CMake 3.1
 # we do manual fallbacks for CMake 3.0 in the compilers section
-set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD 20)
 set(CMAKE_CXX_STANDARD_REQUIRED YES)
 
 set(CMAKE_OSX_DEPLOYMENT_TARGET "10.9" CACHE STRING "Minimum OS X deployment version")
diff -up flightgear-2020.3.18/package/org.flightgear.FlightGear.metainfo.in.omv~ flightgear-2020.3.18/package/org.flightgear.FlightGear.metainfo.in
--- flightgear-2020.3.18/package/org.flightgear.FlightGear.metainfo.in.omv~	2023-06-20 03:33:55.455242456 +0200
+++ flightgear-2020.3.18/package/org.flightgear.FlightGear.metainfo.in	2023-06-20 03:33:57.625257718 +0200
@@ -1,4 +1,3 @@
-
 <?xml version="1.0" encoding="UTF-8"?>
 <!-- Copyright 2014 Richard Hughes <richard@hughsie.com> -->
 <!-- HOUSEKEEPING, REMOVE THIS COMMENT WHEN THIS GOES UPSTREAM
